src/function.jl

```diff
@@ -5,7 +5,10 @@ struct AutoForwardDiff <: AbstractADType end
 struct AutoReverseDiff <: AbstractADType end
 struct AutoTracker <: AbstractADType end
 struct AutoZygote <: AbstractADType end
-struct AutoFiniteDiff <: AbstractADType end
+struct AutoFiniteDiff <: AbstractADType 
+    difftype::Symbol
```

Comment:
We should probably have this as a kwarg to the OptimizationFunction dispatch for it

Reply:
Okay, yeah, I realised that as well. I'll add kwargs `fdtype` and `fdhtype`